### PR TITLE
[SDK-3348] Implement trusted web activity support

### PR DESCRIPTION
**Pull request by poovamraj**
### Changes

> **Warning**
> This is an experimental feature that can be changed in the future

Trusted Web Activity is a feature provided by some browsers to provide a native feel to the custom tabs.
We have implemented this feature to web authentication to provide a more native login experience using web login

```kotlin
WebAuthProvider.login(account)
    .withTrustedWebActivity()
    .await(this)
```

### References
https://developer.chrome.com/docs/android/trusted-web-activity/

### Testing

Please describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. Since this library has unit testing, tests should be added for new functionality and existing tests should complete without errors.

- [x] This change adds unit test coverage

- [x] This change adds integration test coverage

- [x] This change has been tested on the latest version of the platform/language or why not

### CODE REVIEW

File: EXAMPLES.md [hunk 1]
```diff
@@ -119,6 +120,32 @@ WebAuthProvider.logout(account)
     .start(this, logoutCallback)
 ```
 
+## Trusted Web Activity (Experimental)
+> **Warning**
+> Trusted Web Activity support in Auth0.Android is still experimental and can change in the future. 
+>
+> Please test it thoroughly in all the targeted browsers and OS variants and let us know your feedback.
+
+Trusted Web Activity is a feature provided by some browsers to provide a native feel to the custom tabs.
```
Comment (jimmyjames) [inline EXAMPLES.md:129]:
```suggestion
Trusted Web Activity is a feature provided by some browsers to provide a native look and feel to the custom tabs.
```

File: EXAMPLES.md [hunk 2]
```diff
@@ -119,6 +120,32 @@ WebAuthProvider.logout(account)
...
+
+To make it work, there are certain prerequisite
```
Comment (jimmyjames) [inline EXAMPLES.md:131]: To use this feature, there are some additional steps you must take:

File: auth0/src/main/java/com/auth0/android/annotation/ExperimentalAuth0Api.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * The APIs marked with this annotation are considered experimental
+ * The API surface or design could change in future
+ */
+@Retention(CLASS)
+@Target({TYPE, METHOD, CONSTRUCTOR, FIELD, PACKAGE})
+@RequiresOptIn(level = RequiresOptIn.Level.WARNING)
+public @interface ExperimentalAuth0Api {}
```
Comment (jimmyjames) [inline auth0/src/main/java/com/auth0/android/annotation/ExperimentalAuth0Api.java:22]: 🎉 awesome - when I saw the PR included an experimental feature I was going to suggest an annotation denoting the experimental status.

File: auth0/src/main/java/com/auth0/android/provider/CustomTabsController.java [hunk 4]
```diff
@@ -98,29 +107,44 @@ public void unbindService() {
      *
      * @param uri the uri to open in a Custom Tab or Browser.
      */
-    public void launchUri(@NonNull final Uri uri) {
+    public void launchUri(@NonNull final Uri uri, @NonNull final boolean launchAsTwa) {
```
Comment (jimmyjames) [inline auth0/src/main/java/com/auth0/android/provider/CustomTabsController.java:110]: Is this a breaking change, as it's a public method that now requires two non-null parameters?

**Review by jimmyjames — CHANGES_REQUESTED** [review]
Just a few suggestions to the examples, and a question about a change to a public APIs signature.

File: auth0/src/main/java/com/auth0/android/provider/CustomTabsController.java (hunk 4 above)
Comment (poovamraj) [reply auth0/src/main/java/com/auth0/android/provider/CustomTabsController.java:110]: This shouldn't be an issue as CustomTabsController is a package private class and `launchUri` is not exposed outside the library.

I removed the @NonNull annotation as we are using primitive and it doesn't need that annotation

**Review by jimmyjames — APPROVED** [review]

**Pull request merged**